command/ca/provisioner/webhook/update.go

```diff
@@ -0,0 +1,171 @@
+
+## POSITIONAL ARGUMENTS
+
+<name>
+: The name of the webhook.
+
+## EXAMPLES
+
+Change a webhook's url:
+'''
+step ca provisioner webhook update my_webhook --provisioner my_provisioner --url https://example.com
+'''
+
+Configure a webhook to send a bearer token to the server:
+'''
+step ca provisioner webhook update my_webhook --provisioner my_provisioner --bearer-token-file token.txt
+'''
+
+Change the password sent to the webhook with basic authentication:
+'''
+step ca provisioner webhook update my_webhook --provisioner my_provisioner --basic-auth-password-file my_pass.txt
+'''
+
+Configure the webhook to be called only when signing x509 certificates, not SSH certificates:
+'''
+step ca provisioner webhook update my_webhook --provisioner my_provisioner --cert-type X509
+'''
+`,
+	}
+}
+
+func updateAction(ctx *cli.Context) (err error) {
+	if err := errs.NumberOfArguments(ctx, 1); err != nil {
+		return err
+	}
+
+	args := ctx.Args()
+
+	provisionerName := ctx.String("provisioner")
+
+	client, err := newCRUDClient(ctx, ctx.String("ca-config"))
+	if err != nil {
+		return err
+	}
+
+	prov, err := client.GetProvisioner(ca.WithProvisionerName(provisionerName))
+	if err != nil {
+		return err
+	}
+	var wh *linkedca.Webhook
+	for _, pwh := range prov.Webhooks {
+		if pwh.Name == args.Get(0) {
+			wh = pwh
+			break
+		}
+	}
+	if wh == nil {
+		return errors.New("provisioner does not have a webhook with that name")
+	}
+
+	if ctx.IsSet("kind") {
+		kind := linkedca.Webhook_Kind(linkedca.Webhook_Kind_value[ctx.String("kind")])
+		if kind == linkedca.Webhook_NO_KIND {
+			return errors.New("invalid webhook kind")
+		}
+		wh.Kind = kind
+	}
+
+	if ctx.IsSet("url") {
+		wh.Url = ctx.String("url")
+	}
+
+	if ctx.IsSet("bearer-token-file") {
+		bearerTkn, err := utils.ReadStringPasswordFromFile(ctx.String("bearer-token-file"))
+		if err != nil {
+			return err
+		}
+		wh.Auth = &linkedca.Webhook_BearerToken{
+			BearerToken: &linkedca.BearerToken{
+				BearerToken: bearerTkn,
+			},
+		}
+	} else if ctx.IsSet("basic-auth-username") || ctx.IsSet("basic-auth-password-file") {
+		wba, _ := wh.GetAuth().(*linkedca.Webhook_BasicAuth)
+		if wba == nil {
+			wba = &linkedca.Webhook_BasicAuth{
+				BasicAuth: &linkedca.BasicAuth{},
+			}
+		}
+		if wba.BasicAuth == nil {
+			wba.BasicAuth = &linkedca.BasicAuth{}
+		}
+
+		if ctx.IsSet("basic-auth-username") {
+			wba.BasicAuth.Username = ctx.String("basic-auth-username")
+		}
+		if ctx.IsSet("basic-auth-password-file") {
+			password, err := utils.ReadStringPasswordFromFile(ctx.String("basic-auth-password-file"))
+			if err != nil {
+				return err
+			}
+			wba.BasicAuth.Password = password
+		}
+		wh.Auth = wba
+	}
+
+	if ctx.IsSet("disable-tls-client-auth") {
+		wh.DisableTlsClientAuth = ctx.Bool("disable-tls-client-auth")
+	}
+
+	if ctx.IsSet("cert-type") {
+		certType, ok := linkedca.Webhook_CertType_value[ctx.String("cert-type")]
+		if !ok {
+			return errors.New("invalid cert-type")
```

Comment:
Use the errors.InvalidFlagValue error here as well. In general, try to use the errors.xxx errors for bad inputs.

Reply:
97eef2d